hr_timesheet_sheet/models/account_analytic_line.py

```diff
@@ -25,9 +25,13 @@ def _compute_sheet(self):
                 ('company_id', 'in', [timesheet.company_id.id, False]),
                 ('state', 'in', ['new', 'draft']),
             ], limit=1)
-            if timesheet.sheet_id != sheet:
+            if sheet and timesheet.sheet_id != sheet:
```

Comment:
Even when `sheet == False and timesheet.sheet_id != sheet` you should do `timesheet.sheet_id = sheet`.

Reply:
Are you sure? Maybe you are having `timesheet.sheet_id=NewId...` I will try to check...

Reply:
This method is invoked as `res._compute_sheet()` where `res` is the result of a create, so `sheet` is always defined.

Reply:
Sorry, wrong message above :rofl: I will delete it and rewrite it next week :smile:

Reply:
I found a case where sheet was False because the search didn't found a sheet. I will check why/when happens.

Reply:
Exactly! When the search does not find any sheet, `timesheet.sheet_id = sheet` should be done as well.